pinot-transport/src/main/java/com/linkedin/pinot/routing/HelixExternalViewBasedRouting.java

```diff
@@ -32,81 +29,94 @@
 import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
 import com.linkedin.pinot.common.config.TableNameBuilder;
 import com.linkedin.pinot.common.response.ServerInstance;
-import com.linkedin.pinot.common.utils.CommonConstants.Helix.TableType;
+import com.linkedin.pinot.common.utils.CommonConstants;
+import com.linkedin.pinot.common.utils.NetUtil;
 import com.linkedin.pinot.routing.builder.BalancedRandomRoutingTableBuilder;
 import com.linkedin.pinot.routing.builder.KafkaHighLevelConsumerBasedRoutingTableBuilder;
+import com.linkedin.pinot.routing.builder.KafkaLowLevelConsumerRoutingTableBuilder;
 import com.linkedin.pinot.routing.builder.RoutingTableBuilder;
 import com.linkedin.pinot.transport.common.SegmentIdSet;
 
-
-/**
- * HelixExternalViewBasedRouting will maintain the routing table for assigned data table.
- *
- *
- */
 public class HelixExternalViewBasedRouting implements RoutingTable {
-
   private static final Logger LOGGER = LoggerFactory.getLogger(HelixExternalViewBasedRouting.class);
-  private final Set<String> _dataTableSet = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
-  private final RoutingTableBuilder _defaultOfflineRoutingTableBuilder;
-  private final RoutingTableBuilder _defaultRealtimeRoutingTableBuilder;
-  private final Map<String, RoutingTableBuilder> _routingTableBuilderMap;
-
+  private final RoutingTableBuilder _offlineRoutingTableBuilder;
+  private final RoutingTableBuilder _realtimeHLCRoutingTableBuilder;
```

Comment:
agreed.I will add this as a TODO. Hopefully we only need to touch this class (or split it) in order to achieve what you are mentioning. Now that we have the other unused stuff out of the way, that will be easier. We also have tests to make sure we have not broken something.